repair/repair.cc

```diff
@@ -279,6 +280,28 @@ repair_status tracker::get(int id) {
     }
 }
 
+future<repair_status> tracker::get_status_until_finished(int id, std::chrono::steady_clock::time_point timeout) {
+    if (id >= _next_repair_command) {
+        return make_exception_future<repair_status>(std::runtime_error(format("unknown repair id {:d}", id)));
+    }
+    return repeat_until_value([this, id, timeout] {
+        auto it = _status.find(id);
+        if (it == _status.end()) {
+            return make_ready_future<std::optional<repair_status>>(repair_status::SUCCESSFUL);
+        } else {
+            if (it->second == repair_status::FAILED) {
+                return make_ready_future<std::optional<repair_status>>(repair_status::FAILED);
+            } else {
+                return _done_cond.wait(timeout).then([] {
+                    return make_ready_future<std::optional<repair_status>>(std::nullopt);
+                }).handle_exception_type([] (condition_variable_timed_out&) {
+                    return make_ready_future<std::optional<repair_status>>(repair_status::RUNNING);
+                });
+            }
+        }
+    });
+}
```

Comment:
Again need to think about shutdown while this call is running. We should broadcast on the condition variable during shutdown and wait for all calls to return (using async_sharded_service).

Reply:
I will think about it.

Reply:
See above.